include/RAJA/util/Layout.hpp

```diff
@@ -174,10 +190,18 @@ struct LayoutBase_impl<camp::idx_seq<RangeInts...>, IdxLin, StrideOneDim> {
    * @param indices  Indices in the n-dimensional space of this layout
    * @return Linear space index.
    */
+
   template <typename... Indices>
+#if defined(BOUNDS_CHECK)
+  RAJA_INLINE RAJA_HOST_DEVICE IdxLin operator()(
+      Indices... indices) const
+  {
+    BoundsCheck<0>(indices...);
+#else
   RAJA_INLINE RAJA_HOST_DEVICE constexpr IdxLin operator()(
       Indices... indices) const
   {
+#endif
```

Comment:
Would it make more sense to have a BOUNDS_CHECK_constexpr instead of repeating this?

Reply:
Ah, are you saying define BOUNDS_CHECK_constexp as a macro? 
#defined BOUNDS_CHECK_constexp constexpr ?

Reply:
yes, defined as constexpr if BOUNDS_CHECK isn't defined, and nothing if it is.